astronomer/providers/amazon/aws/triggers/s3.py

```diff
@@ -18,95 +16,249 @@ def __init__(
+                        self.inactivity_period,
+                        self.min_objects,
+                        self.previous_objects,
+                        self.inactivity_seconds,
+                        self.allow_delete,
+                        self.last_activity_time,
+                    ):
+
+                        yield TriggerEvent({"status": "success"})
+                        return
+
+        except Exception as e:
+            yield TriggerEvent({"status": "error", "message": str(e)})
+
+    def _get_async_hook(self) -> S3HookAsync:
+        return S3HookAsync(aws_conn_id=self.aws_conn_id, verify=self.hook_params.get("verify"))
+
+
+class S3PrefixSensorTrigger(BaseTrigger):
+    def __init__(
+        self,
+        *,
+        bucket_name: str,
+        prefix: Union[str, List[str]],
+        delimiter: str = '/',
+        aws_conn_id: str = 'aws_default',
+        verify: Optional[Union[str, bool]] = None,
+        **hook_params: Any,
+    ):
+        self.bucket_name = bucket_name
+        self.prefix = [prefix] if isinstance(prefix, str) else prefix
+        self.delimiter = delimiter
+        self.aws_conn_id = aws_conn_id
+        self.verify = verify
+        self.hook_params = hook_params
 
     def serialize(self) -> Tuple[str, Dict[str, Any]]:
         """
-        Serialize S3KeyTrigger arguments and classpath.
+        Serialize S3PrefixSensorTrigger arguments and classpath.
         """
         return (
-            "astronomer.providers.amazon.aws.triggers.s3.S3KeyTrigger",
+            "astronomer.providers.amazon.aws.triggers.s3.S3PrefixSensorTrigger",
             {
                 "bucket_name": self.bucket_name,
-                "bucket_key": self.bucket_key,
-                "wildcard_match": self.wildcard_match,
+                "prefix": self.prefix,
+                "delimiter": self.delimiter,
                 "aws_conn_id": self.aws_conn_id,
+                "verify": self.verify,
                 "hook_params": self.hook_params,
             },
         )
 
-    async def run(self):
+    async def run(self) -> AsyncIterator["TriggerEvent"]:  # type: ignore[override]
         """
         Make an asynchronous connection using S3HookAsync.
         """
         hook = self._get_async_hook()
-        async with await hook.get_client_async() as client:
-            while True:
-                self.log.info("Poking for key : s3://%s/%s", self.bucket_name, self.bucket_key)
-                if await self._check_key(client, self.bucket_name, self.bucket_key, self.wildcard_match):
-                    yield TriggerEvent(True)
+        try:
+            async with await hook.get_client_async() as client:
+                while True:
+                    self.log.info('Poking for prefix : %s in bucket s3://%s', self.prefix, self.bucket_name)
+                    buffer = []
+                    for elem in (
+                        hook._check_for_prefix(client, prefix, self.delimiter, self.bucket_name)
```

Comment:
Isn't this `hook._check_for_prefix(client, prefix, self.delimiter, self.bucket_name)` returning `bool`? Is this for loop iterating over the each prefix?

Reply:
it returns a Coroutine[bool]